src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    /// <summary>
+    /// Runs ApiCompat over different assembly tuples.
+    /// </summary>
+    public class ApiCompatRunner
+    {
+        private static List<(Stream leftAssemblyStream, Stream rightAssemblyStream, string assemblyName, string source, string errorMessage)> queue = new();
+
+        public static IEnumerable<CompatDifference> RunApiCompat()
+        {
+            List<CompatDifference> apiDifferences = new();
+            foreach (var apicompatTuples in queue)
+            {
+                // TODO: Add version check and proper way to display api compat error messages for each tuple.
```

Comment:
FYI, I expect this to change when I add the way to compare one left against multiple rights or the other way around. Also I expect to handle the differences to have an identifier that identifies the tuple that the difference belongs to.